console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,173 @@
+	// to which the plugin needs to connect to.
+	// +kubebuilder:validation:Optional
+	Proxy []ConsolePluginProxy `json:"proxy,omitempty"`
+	// i18n is the configuration of plugin's localization resources.
+	// +kubebuilder:validation:Required
+	I18n ConsolePluginI18n `json:"i18n"`
+}
+
+// ConsolePluginI18n holds information on localization resources that are served by
+// the dynamic plugin.
+type ConsolePluginI18n struct {
+	// enabled indicates if the plugin contains localization resource. If enabled,
+	// i18n namespace, that the plugin contains, will be loaded by the console server.
+	// If i18n is not enabled, no i18n namespaces will be loaded for the plugin.
+	// By default loading of i18n namespace is not enabled.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:default=false
+	Enabled bool `json:"enabled"`
+}
+
+// ConsolePluginProxy holds information on various service types
+// to which console's backend will proxy the plugin's requests.
+type ConsolePluginProxy struct {
+	// type is the type of the console plugin's proxy. Currently only "Service" is supported.
+	// +kubebuilder:validation:Required
+	Type ConsolePluginProxyType `json:"type"`
+	// alias is a proxy name that identifies the plugin's proxy. The alias should be
+	// between 1 and 128 characters and must consist only of alphanumeric characters,
+	// hyphens and underscores. An alias name should be unique per plugin.
+	// The console backend exposes following proxy endpoint:
+	//
+	// /api/proxy/plugin/<plugin-name>/<proxy-alias>/<request-path>?<optional-query-parameters>
+	//
+	// Request example path:
+	//
+	// /api/proxy/plugin/acm/search/pods?namespace=openshift-apiserver
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +kubebuilder:validation:Pattern=`^[A-Za-z0-9-_]+$`
+	Alias string `json:"alias"`
+	// service is an in-cluster Service that the plugin will connect to.
+	// The Service must use HTTPS. The console backend exposes an endpoint
+	// in order to proxy communication between the plugin and the Service.
+	// +kubebuilder:validation:Required
+	Service ConsolePluginProxyServiceConfig `json:"service"`
+	// caCertificate provides the cert authority certificate contents,
+	// in case the proxied Service is using custom service CA.
+	// By default, the service CA bundle provided by the service-ca operator is used.
+	// +kubebuilder:validation:Pattern=`^-----BEGIN CERTIFICATE-----([\s\S]*)-----END CERTIFICATE-----\s?$`
+	// +kubebuilder:validation:Optional
+	CACertificate string `json:"caCertificate,omitempty"`
+	// authorize indicates if the proxied request should contain the logged-in user's
+	// OpenShift access token in the "Authorization" request header. For example:
+	//
+	// Authorization: Bearer sha256~kV46hPnEYhCWFnB85r5NrprAxggzgb6GOeLbgcKNsH0
+	//
+	// By default the access token is not part of the proxied request.
+	// +kubebuilder:validation:Required
```

Comment:
Fields must be optional if they are to be defaulted. Required vs optional is processed before defaulting.

In this case, as this is a bool, you don't need to default to false anyway, as this value is the zero value